crypto/x509/v3_ncons.c

```diff
@@ -192,26 +192,17 @@ static int do_i2r_name_constraints(const X509V3_EXT_METHOD *method,
 
 static int print_nc_ipadd(BIO *bp, ASN1_OCTET_STRING *ip)
 {
-    int i, len;
-    unsigned char *p;
-    p = ip->data;
-    len = ip->length;
-    BIO_puts(bp, "IP:");
-    if (len == 8) {
-        BIO_printf(bp, "%d.%d.%d.%d/%d.%d.%d.%d",
-                   p[0], p[1], p[2], p[3], p[4], p[5], p[6], p[7]);
-    } else if (len == 32) {
-        for (i = 0; i < 16; i++) {
-            BIO_printf(bp, "%X", p[0] << 8 | p[1]);
-            p += 2;
-            if (i == 7)
-                BIO_puts(bp, "/");
-            else if (i != 15)
-                BIO_puts(bp, ":");
-        }
-    } else
-        BIO_printf(bp, "IP Address:<invalid>");
-    return 1;
+    /* should be half of ip->length: len1 == len2 == 4 or len1 == len2 == 16 */
+    int len1 = (ip->length + 1) / 2;
```

Comment:
Why +1 here?

Reply:
The +1 leads to rounding up if and only if `ip->length` is not even.
The calculation ensures that `len1` + `len2` == length.
For instance, if length == 8 then `len1` will be 4 and `len2` will be 4, too.
If length == 7 then `len1` will be 4 and `len2` will be 3.

Reply:
I've just added the following comment:
```
    /* len1 + len2 == ip->length where len1 == len2 + 1 if ip->length is odd */
```

Reply:
Update: 
I've simplified the calculation of `len1` and `len2` and adapted the commenting accordingly.
Anyway the semantics in case `ip->length` is not 2 * 4 or 2 * 16 is not really defined.